Source/JavaScriptCore/bytecompiler/BytecodeGenerator.cpp

```
@@ -533,14 +533,22 @@ BytecodeGenerator::BytecodeGenerator(VM& vm, FunctionNode* functionNode, Unlinke
         break;
     case ConstructorKind::Naked:
         if (!isConstructor()) {
-            emitThrowTypeError("Cannot call a constructor without |new|"_s);
+            String constructorName = functionNode->ident().string();
+            if (!constructorName || constructorName.isEmpty())
+                emitThrowTypeError("Cannot call a constructor without |new|"_s);
+            else
+                emitThrowTypeError(Identifier::fromString(m_vm, makeString("Cannot call a constructor "_s, constructorName, " without |new|"_s)));
```

I think we should use `tryMakeString` etc. to avoid Crash for too long string.

---

@Constellation Thanks, but if an OOM occurs, should we throw an error, or fall back to an error message without using `makeString`?

---

Updated